### PR TITLE
UI: Home page filtering

### PR DESCRIPTION
**Pull request by dallonf**
Proof-of-concept of how we can use React Query + React Router for filtering the posts list.

Let me know if I've gone overboard with the TypeScripting here... actually, I'm pretty sure I _have_, just not sure where 🙃

### CODE REVIEW

File: ui/src/utils/throttleState.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export function useThrottleState<T>(
+  current: T,
+  updateBackingValue: (newValue: T) => void,
+  {
+    timeout = 300,
+    immediatelyUpdateIfFalsey = false,
+  }: { timeout?: number; immediatelyUpdateIfFalsey?: boolean } = {}
+): [T, (newValue: T) => void] {
+  const [value, setValue] = React.useState(current);
```
Comment (dallonf) [inline ui/src/utils/throttleState.ts:11]: Shoutout to Github Copilot, which filled in the entire function at this point and was completely wrong about what I wanted it to do 🤣

**Review by Willdotwhite — APPROVED** [review]

File: ui/src/model/skill.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  "ART_3D",
+  "CODE",
+  "DESIGN_PRODUCTION",
+  "SOUND_MUSIC",
+  "TESTING_SUPPORT",
+  "TEAM_LEAD",
+  "OTHER",
+] as const;
+
+export type Skill = typeof allSkills[number];
```
Comment (Willdotwhite) [inline ui/src/model/skill.ts:12]: Out of interest, why `number` here instead of `string`?

Comment (dallonf) [reply ui/src/model/skill.ts:12]: `allSkills` is an array, so its keys are numbers! Essentially, `typeof allStrings[number]` means "give me the type I would get if I indexed `allStrings` with a number, like `allStrings[0]`"

**Review by runny-yolk — APPROVED** [review]

File: ui/src/model/skill.ts (hunk 2 above)
Comment (Willdotwhite) [reply ui/src/model/skill.ts:12]: Aaaah, of course! There was me thinking it was the value type, wondering if this was a hangover from the bitwise mask 😂

**Pull request merged**